vm/parser.ts

```diff
@@ -759,8 +764,17 @@ module J2ME {
       var s = this;
       var methods_count = s.readU2();
       var m = this.methods = new Array(methods_count);
+      var ids = this.methodIDs = new Array(methods_count);
       for (var i = 0; i < methods_count; i++) {
         m[i] = s.offset;
+
+        ids[i] = [
```

Comment:
> Erm, no, with this change we're going to load eagerly only a subset of data about all the methods of a class: name, signature, and flags. And we have to load this information at startup in order to minimally link virtual methods on Klass prototypes at startup.

We're not loading everything about a method, but IIRC decoding the strings was one of the slowest things in method loading (I'm not sure I recall correctly, @mbebenita did some measurements).
We do need to load this data anyway, you're right (I forgot we need to do it anyway even if we don't have any compiled methods), but I think it would be better to do it only once.

> We don't have to do it this way, of course. We could instead make MethodInfo lighter weight and create lightweight MethodInfo objects for every method in every class we load at startup, as #1225 does. Or this branch could add a MethodInfo.scanMethods2 method that linkKlassMethods calls when it's time to minimally link the methods.
> But then we'd be scanning methods twice on startup: once to retrieve just their indexes, and once to retrieve their name/signature/flags.

I suspect this would be better, as we would avoid loading this data about methods twice. We could scan methods to get their indexes, and then another method could load the name/signature/fields when needed. Or, we could directly load them in MethodInfo in scanMethods, but I'd avoid loading the data twice.